src/Implement/RequestHandler.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "RequestHandler.h"
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <string>
+
+std::unordered_map<std::string, APIHandler *> RequestHandler::apis;
+
+RequestHandler::RequestHandler() {
+  
+}
+
+void RequestHandler::SetFd(const int &file_descriptor){
+  this->client_fd = file_descriptor;
+}
+
+void RequestHandler::WaitForMessage() {
+  std::string tmp_string = "";
+  char input_buffer[4096] = {};
+  std::string next_request = "";
+  // only read header if message is bigger than buffer
+  while (true) {
+    int function_flag =
+        recv(this->client_fd, input_buffer, sizeof(input_buffer), 0);
```

Comment:
MSG_PEEK is available, consider using it instead of accumulating data of the next request in your program.